doc.go

```diff
@@ -3,4 +3,6 @@ package multicodec
 
 //go:generate go run gen.go
 //go:generate gofmt -w code.go
-//go:generate stringer -type=Code -linecomment
+//go:generate go get golang.org/x/tools/cmd/stringer
+//go:generate go run golang.org/x/tools/cmd/stringer -type=Code -linecomment
+//go:generate go mod tidy
```

Comment:
Clever :) I personally prefer to not do hacks like these until https://github.com/golang/go/issues/42088 is fixed (likely in Go 1.17), but having to remember to set up stringer is also a bit of a hack.

I also slightly lean towards not having it this way, since `go generate` editing or tidying the current module seems a bit surprising. No strong opinion, though.

Reply:
i do like that it works and doesn't need stringer on the user path since that's not something I want to always have.
I'll file an issue to revisit once go 1.17 is out, but leave it like this from now.